### PR TITLE
TandemFoil: T_max=10 long training (retry — ensure 14+ epochs)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The previous T_max=10 run (PR #2490) achieved our best TandemFoil result of 75.59 val MAE in only 14 epochs, with the model still clearly improving at cutoff. The run was limited to 14 epochs by the 30-minute timeout. If we can push past 14 epochs, the model should continue improving toward (and potentially below) 70.

The key mechanism: T_max=10 creates ~75 cosine LR cycles per epoch (750 steps/epoch). This extremely rapid LR averaging produces the deepest minima we've observed on TandemFoil. The training curve was still sharply descending at epoch 14.

## Instructions

Run the **exact** baseline config — no changes. The goal is simply to get more epochs than the previous 14-epoch run. The default 30-minute timeout should be respected; this is a retry to confirm the result under normal runtime conditions.

```bash
cd target/icml2026 && python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 3e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --model_slices 64 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --wandb_group tandem-tmax10-longrun-v2
```

## Baseline

**Current best — TandemFoil (PR #2490):**
- `val_primary/surface_pressure_mae`: **75.59**
- `test_primary/surface_pressure_mae`: 72.12
- Per-split test MAE: single_in_dist=72.33, geom_camber_rc=76.01, geom_camber_cruise=70.80, re_rand=69.34
- W&B run: 77yoba65 (T_max=10, 14 epochs, Lion lr=3e-4)

**Other baselines:**
- AirfRANS: `val_primary/surface_mse` = 0.0248 (PR #2556, emma)
- DrivAerML: `val_primary/surface_rel_l2_pct` = 12.96% (PR #2550, violet)

**Target:** Push below 75.59 by training past 14 epochs. Note: `--epochs 999` is MANDATORY — the timeout controls runtime, not epoch count.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT frieren:

## Results

**Run:** w5ja3ii4 — 8 epochs (30-min timeout, SENPAI_MAX_EPOCHS=50)

| Metric | Baseline (PR #2490, 14 ep) | This run (8 ep) | Delta |
|--------|---------------------------|-----------------|-------|
| **val_primary/surface_pressure_mae** | **75.59** | **91.13** (best, ep 7) | +20.6% ❌ |
| test_primary/surface_pressure_mae | 72.12 | 97.88 (ep 8, peak) | +35.7% ❌ |
| test single_in_dist | 72.33 | 113.67 | +57.2% |
| test geom_camber_rc | 76.01 | 107.35 | +41.2% |
| test geom_camber_cruise | 70.80 | 79.81 | +12.7% |
| test re_rand | 69.34 | 90.69 | +30.8% |

**Val progression per epoch (cosine trough epochs in bold):**

| Epoch | val_primary | loss | lr | time (min/ep) |
|-------|-------------|------|-----|---------------|
| **1** | **190.52** | 0.9548 | 0.0 | ~7.0 (incl. init) |
| 2 | 142.98 | 0.4051 | 3e-4 | ~5.4 |
| **3** | **124.31** | 0.3092 | 0.0 | ~4.7 |
| 4 | 120.27 | 0.3012 | 3e-4 | ~3.8 |
| **5** | **99.51** | 0.2662 | 0.0 | ~3.3 |
| 6 | 101.86 | 0.2354 | 3e-4 | ~2.6 |
| **7** | **91.13** | 0.2176 | 0.0 | ~2.2 |
| 8 | 99.68 | 0.2133 | 3e-4 | ~2.2 |

**Command:**
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999 --agent frieren --wandb-name "frieren/tandem-tmax10-longrun-v2" --wandb-group tandem-tmax10-longrun-v2
```

**W&B run:** w5ja3ii4
**Peak GPU memory:** ~30.6 GB (RTX PRO 6000 Blackwell, 98 GB)

### What happened

**Did not beat baseline** — best val=91.13 at epoch 7 vs baseline's 75.59 at epoch 14. Only 8 epochs completed.

**Critical throughput finding: the gap is a cold-start problem, NOT persistent.** Per-epoch throughput improved dramatically during the run: 5.4 → 4.7 → 3.8 → 3.3 → 2.6 → 2.2 → 2.2 min/epoch. By epochs 7-8, throughput matched the baseline's ~2.1 min/epoch. The first 4-5 epochs suffer from data I/O cache warmup overhead, consuming ~20 minutes of the 30-min budget. Once the filesystem cache is warm, training is fast.

**This explains why the baseline run got 14 epochs:** the baseline run (77yoba65) likely had warm filesystem caches from a preceding run, giving it ~2.1 min/epoch from the start. Our runs all start cold.

**Across all 3 attempts:**
| Run | Epochs | Best val (trough) | Avg min/ep | Final min/ep |
|-----|--------|-------------------|------------|--------------|
| yw8j6ic2 | 8 | 96.39 (ep 8) | 3.75 | — |
| m98jm0ds | 9 | 88.62 (ep 9) | 3.33 | — |
| w5ja3ii4 | 8 | 91.13 (ep 7) | 3.75 | 2.2 |

**Note:** Test metrics are from epoch 8 (lr=3e-4 peak), which is always worse than the trough. The best test would have been at epoch 7 (trough) but final-epoch test eval lands on the peak.

### Suggested follow-ups

1. **Pre-warm the filesystem cache** — Run a quick throwaway epoch or `cat` the training data files before the actual training run. This could recover ~10 min of wasted I/O time and add 4-5 more epochs.
2. **Back-to-back runs** — Start a second run immediately after the first finishes (cache will be warm). The second run should start at ~2.2 min/epoch from epoch 1, giving ~13-14 epochs in 30 min.
3. **The hypothesis remains valid** — trough progression (190→124→99→91) confirms T_max=10 rapid cycling works. The blocker is purely compute: we need 14+ epochs to beat the baseline, and cold-start overhead prevents that in a single 30-min window.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2604 because ADVISOR: Closing after 3 failed attempts — all hit the same cold-start filesystem I/O wall (first 4-5 epochs ~5.4 min/ep instead of ~2.2 min/ep), limiting training to 8-9 epochs when we need 14+. The original 77yoba65 run had warm caches. The hypothesis is sound but unproven under current infrastructure. Reassigning frieren to a different direction rather than chase the same wall again..
